LibreNMS/Modules/Slas.php

```diff
@@ -0,0 +1,614 @@
+<?php
+/**
+ * SLA.php
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * @link       https://www.librenms.org
+ */
+
+namespace LibreNMS\Modules;
+
+use App\Models\Sla;
+use LibreNMS\Interfaces\Module;
+use LibreNMS\OS;
+use LibreNMS\RRD\RrdDefinition;
+use LibreNMS\Util\IP;
+
+class Slas implements Module
+{
+    //use SyncsModels;
+
+    /**
+     * Discover this module. Heavier processes can be run here
+     * Run infrequently (default 4 times a day)
+     *
+     * @param OS $os
+     */
+    public function discover(OS $os)
+    {
+        $device = $os->getDeviceArray();
+
+        if ($device['os_group'] == 'cisco') {
+            $this->discoverSlas($device);
+        } elseif ($device['os'] == 'junos') {
+            $this->discoverRpms($device);
+        }
+        // } else if ($device['os'] == 'huawei')
+        // {
+        //    $this->discoverHuaweis($device);
+        // }
+    }
+
+    /**
+     * Poll data for this module and update the DB / RRD.
+     * Try to keep this efficient and only run if discovery has indicated there is a reason to run.
+     * Run frequently (default every 5 minutes)
+     *
+     * @param OS $os
+     */
+    public function poll(OS $os)
+    {
+        $device = $os->getDeviceArray();
+
+        if ($device['os_group'] == 'cisco') {
+            $this->pollSlas($device);
+        } elseif ($device['os'] == 'junos') {
+            $this->pollRpms($device);
+        }
+        // } else if ($device['os'] == 'huawei')
+        // {
+        //    $this->pollHuaweis($device);
+        // }
+    }
+
+    /**
+     * Remove all DB data for this module.
+     * This will be run when the module is disabled.
+     *
+     * @param OS $os
+     */
+    public function cleanup(OS $os)
+    {
+        $os->getDevice()->printerSupplies()->delete();
+    }
+
+    private function discoverRpms($device)
+    {
+        $slas = snmp_walk($device, 'pingMIB.pingObjects.pingCtlTable.pingCtlEntry', '-OQUs', '+DISMAN-PING-MIB');
```

Comment:
Hi @geg347 
DISMAN is indeed a standard mib, you are right. Meaning other vendors could support it. In that case, instead of moving DISMAN code to Juniper class, it would make sense to keep it here. 
Cisco code, on the other end, should move to specific Cisco OS classes. 
And Huawei (VRP) is also using proprietary so I'll use the class model as well.

In term of discovery and polling : 
- Discovery should try to call OS and standard (DISMAN) 
- Polling should poll only whatever was discovered -> best performance.